src/components/bottom/oneclick-daily-digest/main.js

```
@@ -0,0 +1,56 @@
+import myft from 'next-myft-client';
+import Cookies from 'js-cookie';
+
+module.exports = function customSetup (banner, done) {
+	if (window.FT && window.FT.flags && window.FT.flags.oneClickDailyDigest) {
+		function loadSuccessContent () {
+			const outerContainer = banner.bannerElement.querySelector('.o-banner__outer');
+			outerContainer.classList.add('o-banner_success-background');
+			const closeButton = banner.bannerElement.querySelector('.o-banner__close');
+			closeButton.classList.add('o-banner__close-color');
+			const signupContent = banner.bannerElement.querySelector('.o-banner_signup-content');
+			signupContent.classList.add('--is-visible');
+			const successContent = banner.bannerElement.querySelector('.o-banner_success-content');
+			successContent.classList.remove('--is-visible');
+		}
+
+		function handleSignUpClick (evt) {
+			evt.preventDefault();
+
+			function getCSRFToken () {
+				const desiredTokenLength = 36;
+				const token = Cookies.get('FTSession_s') || Cookies.get('FTSession');
+				const csrfToken = token ? token.slice(-desiredTokenLength) : '';
+				return csrfToken;
+			}
+
+			function addUserToDigest () {
+				try {
+					return myft.add('user', null, 'preferred', 'preference', 'email-digest', {
+						token: getCSRFToken,
+						_rel:{
+							type: 'daily',
+							sendTime:'every morning'
+						}
+					});
+				} catch(e) {
+					return e;
```

If this was an async function - if you return `e` here, the promise chain assumes that the error has been handled and will never go into your catch block - either log and re-throw the error or don't catch it and let it bubble up.

On a side note - since this isn't marked as an `async` function, what here would through a synchronous error?